Std/Tactic/Alias.lean

```diff
@@ -0,0 +1,165 @@
+The `alias` command is used to create synonyms. The plain command can create a synonym of any
+declaration. There is also a version to create synonyms for the forward and reverse implications of
+an iff theorem.
+-/
+
+namespace Std.Tactic.Alias
+
+open Lean Elab Parser.Command
+
+/-- An alias can be in one of three forms -/
+inductive AliasInfo where
+  /-- Plain alias -/
+  | plain (n : Name)
+  /-- Forward direction of an iff alias -/
+  | forward (n : Name)
+  /-- Reverse direction of an iff alias -/
+  | reverse (n : Name)
+deriving Inhabited
+
+/-- The name underlying an alias target -/
+def AliasInfo.name : AliasInfo → Name
+  | plain n => n
+  | forward n => n
+  | reverse n => n
+
+/-- The docstring for an alias. -/
+def AliasInfo.toString : AliasInfo → String
+  | plain n => s!"**Alias** of `{n}`."
+  | forward n => s!"**Alias** of the forward direction of `{n}`."
+  | reverse n => s!"**Alias** of the reverse direction of `{n}`."
+
+
+/-- Environmant extension for registering aliases -/
+initialize aliasExt : SimpleScopedEnvExtension (Name × AliasInfo) (NameMap AliasInfo) ←
+  registerSimpleScopedEnvExtension {
+    addEntry := fun st (n, i) => st.insert n i
+    initial := {}
+  }
+
+/-- Get the alias information for a name -/
+def getAliasInfo [Monad m] [MonadEnv m] (name : Name) : m (Option AliasInfo) := do
+  return aliasExt.getState (← getEnv) |>.find? name
+
+/-- Set the alias info for a new declaration -/
+def setAliasInfo [MonadEnv m] (info : AliasInfo) (declName : Name) : m Unit :=
+  modifyEnv (aliasExt.addEntry · (declName, info))
+
+/--
+  The command `alias name := target` creates a synonym of `target` with the given name.
+
+  The command `alias ⟨fwd, rev⟩ := target` creates synonyms for the forward and reverse directions
+  of an iff theorem. Use `_` if only one direction is required.
+
+  These commands accept all modifiers and attributes that `def` and `theorem` do.
+ -/
+elab (name := alias) mods:declModifiers "alias " alias:ident " := " name:ident : command =>
+  Command.liftTermElabM do
+    let resolved ← resolveGlobalConstNoOverloadWithInfo name
+    let const ← getConstInfo resolved
+    let declMods ← elabModifiers mods
+    let (declName, _) ← mkDeclName (← getCurrNamespace) declMods alias.getId
+    let decl : Declaration := match const with
+      | Lean.ConstantInfo.thmInfo t =>
+        .thmDecl { t with
+          name := declName
+          value := mkConst resolved (t.levelParams.map mkLevelParam)
+        }
+      | Lean.ConstantInfo.defnInfo c
+      | Lean.ConstantInfo.quotInfo c
+      | Lean.ConstantInfo.inductInfo c
+      | Lean.ConstantInfo.axiomInfo c
+      | Lean.ConstantInfo.opaqueInfo c
+      | Lean.ConstantInfo.ctorInfo c
+      | Lean.ConstantInfo.recInfo c =>
+        .defnDecl { c with
+          name := declName
+          value := mkConst resolved (c.levelParams.map mkLevelParam)
+          hints := .regular 0 -- FIXME
+          safety := if declMods.isUnsafe then .unsafe else .safe
```

Comment:
```suggestion
          safety := if declMods.isUnsafe then .unsafe else c.safety
```
I don't think it should be necessary to use `unsafe alias` to alias an unsafe declaration, it can just infer the right thing.

Reply:
My thought process here and below was to use the same behavior as `def` so that users "instinctively" know how to use `alias` but I'm happy to change that.

Reply:
the idea is that `alias A := B` should always "just work" for a wide variety of choices of `B`, this is one of the advantages of `alias` over `def` (you can get alias-like behavior from def using `def A := @B`).